app/controllers/services_controller.rb

```diff
@@ -42,6 +42,17 @@ def services_for_user
     render json: ServicesSerializer.new(services).attributes, status: :ok
   end
 
+  def destroy
+    service_id = params[:id]
+    if Service.where(id: service_id).empty?
+      head :bad_request
+    else
+      Service.find(service_id).destroy!
```

Comment:
I think by destroying the service the associated metadata is also destroyed? As it is declared as:
has_many :metadata, dependent: :destroy

Reply:
I think I tested it and it didn't. But I will retest and see.